pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -391,10 +394,13 @@ func (f *framework) RunPreFilterExtensionRemovePod(
 	podToRemove *v1.Pod,
 	nodeInfo *schedulernodeinfo.NodeInfo,
 ) (status *Status) {
-	startTime := time.Now()
-	defer func() {
-		metrics.FrameworkExtensionPointDuration.WithLabelValues(preFilterExtensionRemovePod, status.Code().String()).Observe(metrics.SinceInSeconds(startTime))
-	}()
+	// This extension point is run for multiple nodes in a scheduling cycle. Sample it to avoid impacting performance.
+	if state.ShouldRecordFrameworkMetrics() {
+		startTime := time.Now()
+		defer func() {
+			metrics.FrameworkExtensionPointDuration.WithLabelValues(preFilterExtensionRemovePod, status.Code().String()).Observe(metrics.SinceInSeconds(startTime))
```

Comment:
As you mentioned, it is going to be confusing that this extension point is measured per node, while the score is for all nodes. 

Lets measure this extension point outside in generic_scheduler.go (specifically in ```findNodesThatFitPod``` measure ```findNodesThatPassFilters```).

Reply:
We can do this. But it's a bit confusing too: the filter extension point latency isn't for the `RunFilterPlugins` method, while other extension point metrics implicitly equals to the latency of each `RunXXPlugin` method.

Reply:
But it's probably better than introducing a new metric. Let me give it a try.

Reply:
There is another issue. The `RunPreFilterExtensionAddPod` method is also invoked as part of `findNodesThatPassFilters`. Do you think it's OK to include this latency in framework extension point latency for `Filter`?

Reply:
I think this is fine. The added latency should be minimal since we don't expect to have many nominated pods at a time.